--- a/C/Aes.c
+++ b/C/Aes.c
@@ -53,6 +53,7 @@
 
 // #define _SHOW_AES_STATUS
 
+#ifndef Z7_NO_HW_AES
 #ifdef MY_CPU_X86_OR_AMD64
   #define USE_HW_AES
 #elif defined(MY_CPU_ARM_OR_ARM64) && defined(MY_CPU_LE)
@@ -70,8 +71,9 @@
     #endif
   #endif
 #endif
+#endif
 
-#ifdef USE_HW_AES
+#if defined(USE_HW_AES) && !defined(Z7_NO_HW_AES)
 #ifdef _SHOW_AES_STATUS
 #include <stdio.h>
 #define _PRF(x) x
@@ -122,7 +124,7 @@
   UInt32 flags = 0;
   #endif
   
-  #ifdef USE_HW_AES
+  #if defined(USE_HW_AES) && !defined(Z7_NO_HW_AES)
   if (CPU_IsSupported_AES())
   {
     // #pragma message ("AES HW")

