--- a/CPP/7zip/Bundles/Format7zF/Arc_gcc.mak
+++ b/CPP/7zip/Bundles/Format7zF/Arc_gcc.mak
@@ -31,18 +31,6 @@
   $O/OutMemStream.o \
   $O/ProgressMt.o \
   $O/Threads.o \
-  $O/lz4-mt_common.o \
-  $O/lz4-mt_compress.o \
-  $O/lz4-mt_decompress.o \
-  $O/brotli-mt_common.o \
-  $O/brotli-mt_compress.o \
-  $O/brotli-mt_decompress.o \
-  $O/lizard-mt_common.o \
-  $O/lizard-mt_compress.o \
-  $O/lizard-mt_decompress.o \
-  $O/lz5-mt_common.o \
-  $O/lz5-mt_compress.o \
-  $O/lz5-mt_decompress.o \
 
 endif
 
@@ -144,10 +132,6 @@
   $O/XarHandler.o \
   $O/XzHandler.o \
   $O/ZHandler.o \
-  $O/ZstdHandler.o \
-  $O/LizardHandler.o \
-  $O/Lz5Handler.o \
-  $O/Lz4Handler.o \
 
 AR_COMMON_OBJS = \
   $O/CoderMixer2.o \
@@ -282,23 +266,6 @@
   $O/ZlibDecoder.o \
   $O/ZlibEncoder.o \
   $O/ZDecoder.o \
-  $O/ZstdDecoder.o \
-  $O/ZstdEncoder.o \
-  $O/ZstdRegister.o \
-  $O/Lz4Decoder.o \
-  $O/Lz4Encoder.o \
-  $O/Lz4Register.o \
-  $O/BrotliDecoder.o \
-  $O/BrotliEncoder.o \
-  $O/BrotliRegister.o \
-  $O/LizardDecoder.o \
-  $O/LizardEncoder.o \
-  $O/LizardRegister.o \
-  $O/Lz5Decoder.o \
-  $O/Lz5Encoder.o \
-  $O/Lz5Register.o \
-  $O/FastLzma2Register.o \
-  $O/LzhamRegister.o \
 
 ifdef DISABLE_RAR
 DISABLE_RAR_COMPRESS=1
@@ -375,11 +342,9 @@
   $O/7zCrc.o \
   $O/7zCrcOpt.o \
   $O/Aes.o \
-  $O/AesOpt.o \
+  $O/AesHwStub.o \
   $O/Sha256.o \
-  $O/Sha256Opt.o \
   $O/Sha1.o \
-  $O/Sha1Opt.o \
 
 ZSTD_LIB = $O/libzstd.$(DYSUFFIX)
 LZ4_LIB = $O/liblz4.$(DYSUFFIX)
@@ -394,19 +359,6 @@
   $O/liblzhamcomp.$(DYSUFFIX)
 
 ADDED_HASH_OBJS = \
-  $O/md2.o \
-  $O/md4.o \
-  $O/md5.o \
-  $O/sha512.o \
-  $O/blake3.o \
-  $O/Md2Reg.o \
-  $O/Md4Reg.o \
-  $O/Md5Reg.o \
-  $O/Sha384Reg.o \
-  $O/Sha512Reg.o \
-  $O/XXH32Reg.o \
-  $O/XXH64Reg.o \
-  $O/Blake3Reg.o \
 
 ARC_OBJS = \
   $(LZMA_DEC_OPT_OBJS) \

